pkg/gui/controllers/global_controller.go

```diff
@@ -37,10 +39,12 @@ func (self *GlobalController) customCommand() error {
 		Title:               self.c.Tr.CustomCommand,
 		FindSuggestionsFunc: self.GetCustomCommandsHistorySuggestionsFunc(),
 		HandleConfirm: func(command string) error {
-			self.c.GetAppState().CustomCommandsHistory = utils.Limit(
-				lo.Uniq(append(self.c.GetAppState().CustomCommandsHistory, command)),
-				1000,
-			)
+			if !strings.HasPrefix(command, " ") {
```

Comment:
I'd extract out `!strings.HasPrefix(command, " ")` into a method called `shouldSaveCommand(command)` to make things clearer, and I'd add a comment explaining that this mimics existing behaviour in shells (I didn't realise this upon first reviewing).